### ISSUE TITLE
tweetnacl-tools 

### ISSUE THREAD

**Issue by jv813yh** [sbp/tweetnacl-tools#2]
I want to ask if you can help me with this problem. I downloaded your tweetnacl-tools program and wanted to test it on windows 10 / Ubuntu 20.04 LTS. But on windows, if I enter a display such as tweetnacl-keypair alice.pub alice.sec, nothing is done. And when I tried the same display on Ubuntu it showed me that the display was not found. Can you advise me where I make a mistake?

**Comment by sbp** [comment]
You have to [compile](https://en.wikipedia.org/wiki/Compiler) the programs first, as noted in the [README file](https://github.com/sbp/tweetnacl-tools/blob/master/README.md):

> Type `make` to compile the tools using `gcc`. The scripts go in `bin/`.

And then to run them in a standard Linux shell like [bash](https://en.wikipedia.org/wiki/Bash_%28Unix_shell%29) you either have to do e.g. `bin/tweetnacl-keypair [...]` in the `tweetnacl-tools` directory that you cloned or downloaded, or change directory into `bin/` and then run `./tweetnacl-keypair [...]`.

In other words you have to give the full path to the executable, which is the name for the file which is the output of the compilation process. If you want to type just `tweetnacl-keypair` the program's executable would have to be on what's called your [shell path](https://ostoday.org/linux/how-to-check-path-in-linux.html). The aforelinked article gives some tips on how to accomplish that.

Many distributions of software come with a program that installs the executables onto the shell path for the user automatically, but *tweetnacl-tools* assumes, perhaps incorrectly, some level of technical knowledge due to its situation as a niche cryptographic suite of tools.

**Comment by jv813yh** [comment]
Oh well, thank you, I'm fine.I would like to ask one more question, namely that the TweetNacl library is also functional on the Windows operating system.
________________________________
Od: Sean B. Palmer <notifications@github.com>
Odoslané: štvrtok 29. októbra 2020 9:33
Komu: sbp/tweetnacl-tools <tweetnacl-tools@noreply.github.com>
Kópia: Jozef Vendeľ <jozef.vendel@student.tuke.sk>; Author <author@noreply.github.com>
Predmet: Re: [sbp/tweetnacl-tools] tweetnacl-tools (#2)


You have to compile<https://en.wikipedia.org/wiki/Compiler> the programs first, as noted in the README file<https://github.com/sbp/tweetnacl-tools/blob/master/README.md>:

Type make to compile the tools using gcc. The scripts go in bin/.

And then to run them in a standard Linux shell like bash<https://en.wikipedia.org/wiki/Bash_%28Unix_shell%29> you either have to do e.g. bin/tweetnacl-keypair [...] in the tweetnacl-tools directory that you cloned or downloaded, or change directory into bin/ and then run ./tweetnacl-keypair [...].

In other words you have to give the full path to the executable, which is the name for the file which is the output of the compilation process. If you want to type just tweetnacl-keypair the program's executable would have to be on what's called your shell path<https://ostoday.org/linux/how-to-check-path-in-linux.html>. The aforelinked article gives some tips on how to accomplish that.

Many distributions of software come with a program that installs the executables onto the shell path for the user automatically, but tweetnacl-tools assumes, perhaps incorrectly, some level of technical knowledge due to its situation as a niche cryptographic suite of tools.

**Comment by jv813yh** [comment]
I would like to ask if this program is also functional on the Windows 10 operating system. Or if you do not have a project that uses the TweetNacl library on the Windows operating system.
